bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/MenuItem.java

```diff
@@ -1121,27 +1124,38 @@ LRESULT wmCommandChild (long wParam, long lParam) {
 	return null;
 }
 
-LRESULT wmDrawChild (long wParam, long lParam) {
-	DRAWITEMSTRUCT struct = new DRAWITEMSTRUCT ();
-	OS.MoveMemory (struct, lParam, DRAWITEMSTRUCT.sizeof);
-	if (image != null) {
+LRESULT wmDrawChild(long wParam, long lParam) {
+	DRAWITEMSTRUCT struct = new DRAWITEMSTRUCT();
+	OS.MoveMemory(struct, lParam, DRAWITEMSTRUCT.sizeof);
+	if ((text != null || image != null)) {
 		GCData data = new GCData();
 		data.device = display;
 		GC gc = createNewGC(struct.hDC, data);
-		/*
-		* Bug in Windows.  When a bitmap is included in the
-		* menu bar, the HDC seems to already include the left
-		* coordinate.  The fix is to ignore this value when
-		* the item is in a menu bar.
-		*/
-		int x = (parent.style & SWT.BAR) != 0 ? MARGIN_WIDTH * 2 : struct.left;
-		Image image = getEnabled () ? this.image : new Image (display, this.image, SWT.IMAGE_DISABLE);
+
+		int x = (parent.style & SWT.BAR) == 0 ? MARGIN_WIDTH * 2 : struct.left;
 		int zoom = getZoom();
-		gc.drawImage (image, DPIUtil.scaleDown(x, zoom), DPIUtil.scaleDown(struct.top + MARGIN_HEIGHT, zoom));
-		if (this.image != image) image.dispose ();
-		gc.dispose ();
+		if (text != null) {
+			int flags = SWT.DRAW_MNEMONIC | SWT.DRAW_DELIMITER;
+			boolean isInactive = ((struct.itemState & OS.ODS_INACTIVE) != 0);
+			boolean isSelected = ((struct.itemState & OS.ODS_SELECTED) != 0);
+			gc.setForeground(isInactive? display.getSystemColor(SWT.COLOR_GRAY) :  display.getSystemColor(SWT.COLOR_WHITE));
+			gc.setBackground(isSelected? display.getSystemColor(SWT.COLOR_BLUE) : parent.getBackground());
```

Comment:
While in general I like the new colors more than the old ones (in particular the white instead of the gray text in a selected entry), the region that is colored is incorrect. Only the label is affected, but actually the background of the whole menu item should be drawn in that color, as it was done before. Maybe you can simply fill the whole menu item with that color first? And maybe it is also possible to use the old blue tone for that?
Here is how a selected item looks now:
![Screenshot 2024-08-19 130943](https://github.com/user-attachments/assets/f3e6f796-7c01-4241-ba18-df21d6f46041)

In here is how it looked before:
![Screenshot 2024-08-19 130933](https://github.com/user-attachments/assets/0280da92-ee26-4b15-b0eb-d4319083b20d)

You can also see that the text color has changed in general (from a grayish tone to white), which I personally prefer, but we have to consider that this is an additional change.